### ISSUE TITLE
Websocket connection failure

### ISSUE THREAD

**Issue by mesqueeb** [INFURA/infura#176]
I get an error trying to connect to the websocket like so:
> index.js?3c3c:42 WebSocket connection to 'wss://kovan.infura.io/ws/v3/myid' failed: Received a broken close frame containing a reserved status code.
subscribe error: Error: CONNECTION ERROR: Couldn't connect to node on WS.
    at Object.InvalidConnection (errors.js?5f2e:35)
    at WebsocketProvider._timeout (index.js?3c3c:184)
    at WebSocket.connection.onerror (index.js?3c3c:92)

I changed `myid` above with my project id

**Comment by JeeChoi** [comment]
Hello, 

The Kovan websocket endpoint is up and running. Are you using a v3 project ID? A v3 project ID is a project ID that was generated by the Infura Dashboard after July of 2018. For more information about Infura Dashboard: https://blog.infura.io/infura-dashboard-update-9f02d0643eb3

We also have https://community.infura.io/ where users can ask questions about our service.

Let me know if you need more help.
Jee

**Comment by mesqueeb** [comment]
I will post again if it stops working. It seems to work now!

**Comment by mesqueeb** [comment]
BTW, @JeeChoi is it ok to expose your Project-ID to your front-end?

**Comment by JeeChoi** [comment]
Hi, it's ok to expose your Project-ID but do not expose your project secret. 

more detailed answer: https://community.infura.io/t/is-it-ok-to-expose-my-infura-project-id-while-connecting-via-web3js-on-the-client-side/45

**Comment by rubenrdp** [comment]
I am also having this same issue (Received a broken close frame containing a reserved status code.) when invoking a contract function using the method `send()` from Web3.js (1.0.0). 

I tried to use 'wss://ropsten.infura.io/ws' as provider (also 'wss://ropsten.infura.io/ws/v3/my_ID' after I saw this thread) when instantiating web3. This works OK but after I call `send()` I keep getting this error message.

I opened this issue https://github.com/INFURA/infura/issues/179 but maybe both can be grouped here.

Thanks
